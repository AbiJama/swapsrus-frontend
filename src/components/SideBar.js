import React, { useState } from "react";
import "../styles/sidebar.css";

function SideBar({ handleSetToys }) {
  const [selectedCategory, setSelectedCategory] = useState("");
  const [isOpen, setIsOpen] = useState(false);

  const handleClick = (category) => {
    setSelectedCategory(category);
    handleSetToys(category);
  };

  const toggleSidebar = () => {
    setIsOpen(!isOpen);
  };

  const clearFilter = () => {
    setSelectedCategory("");
    handleSetToys("all");
  };

  console.log(selectedCategory);

  return (
    <div className={`sidebar ${isOpen ? "open" : ""}`}>
      <button type="button" className="toggle-button" onClick={toggleSidebar}>
        {isOpen ? "Close Sidebar" : "Open Sidebar"}
      </button>

      {isOpen && (
        <>
          <h1 className="sidebar-heading">Filter by type</h1>
          <ul className="sidebar-links">
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Books")}
            >
              Books
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Pre-school")}
            >
              Pre-school
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Indoor")}
            >
              Indoor
            </button>
            <button
              type="button"
              className={`sidebar-links-item ${selectedCategory === "Outdoor" ? "selected" : ""}`}
              onClick={() => handleClick("Outdoor")}
            >
              Outdoor
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={clearFilter}
            >
              Clear Filter
            </button>
          </ul>
          <h2 className="sidebar-heading">Filter by Condition</h2>
          <ul className="sidebar-links">
          <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Brand New")}
            >
              Brand New
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Like New")}
            >
              Like New
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Good")}
            >
              Good
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Used")}
            >
              Used
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={() => handleClick("Defected")}
            >
              Defected
            </button>
            <button
              type="button"
              className="sidebar-links-item"
              onClick={clearFilter}
            >
              Clear Filter
            </button>
          </ul>
        </>
      )}
    </div>
  );
}

export default SideBar;
