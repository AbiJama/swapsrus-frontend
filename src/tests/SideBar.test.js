import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SideBar from "../components/SideBar";

describe("SideBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SideBar />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Filter By Type/i)).toBeInTheDocument();
    expect(screen.getByText(/Clear Filter/i)).toBeInTheDocument();
    expect(screen.getByText(/AgeRange Ascending/i)).toBeInTheDocument();
    expect(screen.getByText(/AgeRange Descending/i)).toBeInTheDocument();
    expect(screen.getByText(/Clear Sort/i)).toBeInTheDocument();
  });
});
