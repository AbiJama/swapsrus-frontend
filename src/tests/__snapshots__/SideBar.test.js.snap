// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="sidebar-heading"
    >
      Filter by type
    </h3>
    <ul
      class="sidebar-links"
    >
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Books
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Pre-school
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Indoor
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item "
          type="button"
        >
          Outdoor
        </button>
      </li>
      <li>
        <button
          class="clear sidebar-links-item"
          type="button"
        >
          Clear Filter
        </button>
      </li>
    </ul>
  </div>
  <div>
    <h3
      class="sidebar-heading"
    >
      Filter by Condition
    </h3>
    <ul
      class="sidebar-links"
    >
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Brand New
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Like New
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Good
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Fair
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          Defected
        </button>
      </li>
      <li>
        <button
          class="clear sidebar-links-item"
          type="button"
        >
          Clear Filter
        </button>
      </li>
    </ul>
  </div>
  <div>
    <h3
      class="sidebar-heading"
    >
      Filter by Age Range
    </h3>
    <ul
      class="sidebar-links"
    >
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          0-3
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          3-6
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          6-9
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          9-12
        </button>
      </li>
      <li>
        <button
          class="sidebar-links-item"
          type="button"
        >
          12+
        </button>
      </li>
      <li>
        <button
          class=" clear sidebar-links-item"
          type="button"
        >
          Clear Filter
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
